testsuite/tests/parallel/domain_id.ml

```diff
@@ -12,9 +12,9 @@ let id () = ()
 
 let newdom_id () =
   let d = Domain.spawn id in
-  let n = Domain.get_id d in
+  let n = Domain.get_id d |> string_of_id |> int_of_string in
```

Comment:
```ocaml
Domain.get_id d |> string_of_id |> int_of_string
```

I'm not comfortable with this pattern. `string_of_id` doesn't guarantee that it returns the stringified version of an integer. If all the uses need this pattern, we really should have `int_of_id` instead of `string_of_id`. We can guarantee uniqueness in the API doc

```
For two different domains that are currently running, the integer form of their
domain ids will be distinct. That is, if [id1] and [id2] the domain ids of the
current running domains, then [not (int_of_id id1 = int_of_id id2)] holds.
```